DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs

```diff
@@ -17,33 +18,49 @@ public class SettingController : UIEngineController
+            string Host_ApiKey = HostController.Instance.GetEncryptedString("Vanjaro.Integration.YouTube.ApiKey", Config.GetDecryptionkey());
 
-            Settings.Add("ApiKey", new UIData { Name = "ApiKey", Value = apikey });
-            Settings.Add("HasApiKey", new UIData { Name = "HasApiKey", Options = HasApiKey });
+            Settings.Add("IsSuperUser", new UIData { Name = "IsSuperUser", Options = UserController.Instance.GetCurrentUserInfo().IsSuperUser });
+            Settings.Add("ApplyTo", new UIData { Name = "ApplyTo", Options = false });
+            Settings.Add("Host_ApiKey", new UIData { Name = "Host_ApiKey", Value = Host_ApiKey });
+            Settings.Add("Host_HasApiKey", new UIData { Name = "Host_HasApiKey", Options = string.IsNullOrEmpty(Host_ApiKey) ? false : true });
+            Settings.Add("Site_ApiKey", new UIData { Name = "Site_ApiKey", Value = Site_ApiKey });
+            Settings.Add("Site_HasApiKey", new UIData { Name = "Site_HasApiKey", Options = string.IsNullOrEmpty(Site_ApiKey) ? false : true });
             return Settings.Values.ToList();
         }
 
         [HttpPost]
         public bool Save(dynamic Data)
         {
-            if (Core.Providers.Youtube.IsValid(Data.ToString()))
+            if (bool.Parse(Data.ApplyTo.ToString()))
             {
-                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube", Data.ToString(), Config.GetDecryptionkey());
-                return true;
+                if (Core.Providers.Youtube.IsValid(Data.Host_ApiKey.ToString()))
+                {
+                    HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.YouTube.ApiKey", Data.Host_ApiKey.ToString(), Config.GetDecryptionkey());
+                }
+                else
+                    return false;
             }
-            return false;
+            else
+            {
+                if (Core.Providers.Youtube.IsValid(Data.Site_ApiKey.ToString()))
+                {
+                    PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube.ApiKey", Data.Site_ApiKey.ToString(), Config.GetDecryptionkey());
+                }
+                else
+                    return false;
+            }
+            return true;
         }
 
-        [HttpGet]
-        public string Delete()
+        [HttpPost]
+        public string Delete(dynamic Data)
         {
-            PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube", string.Empty, Config.GetDecryptionkey());
+            if (bool.Parse(Data.ToString()))
+                HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.YouTube.ApiKey", string.Empty, Config.GetDecryptionkey());
+            else
+                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube.ApiKey", string.Empty, Config.GetDecryptionkey());
```

Comment:
why are you change the key name **Vanjaro.Integration.YouTube** to **this Vanjaro.Integration.YouTube.ApiKey**

Reply:
I believe this (Vanjaro.Integration.YouTube.ApiKey) make more sense that what key we are going to save

Reply:
YouTube Key has been changed